src/stanza.c

```diff
@@ -835,6 +835,40 @@ xmpp_stanza_t *xmpp_stanza_get_child_by_ns(xmpp_stanza_t * const stanza,
     return child;
 }
 
+/** Get the first child of stanza with name and a given namespace.
+ *  This function searches all the immediate children of stanza for a child
+ *  stanza that matches the name and namespace provided.
+ *  The first matching child is returned.
+ *  
+ *  @param stanza a Strophe stanza object
+ *  @param name a string with the name to match
+ *  @param ns a string with the namespace to match
+ *
+ *  @return the matching child stanza object or NULL if no match was found
+ *
+ *  @ingroup Stanza
+ */
+xmpp_stanza_t *xmpp_stanza_get_child_by_name_and_ns(xmpp_stanza_t * const stanza, 
+                                             const char * const name,
+                                             const char * const ns)
+{
+    xmpp_stanza_t *child;
+    const char *child_ns;
+    const char *child_name;
+
+    for (child = stanza->children; child; child = child->next) {
+        child_name = xmpp_stanza_get_name(child);
```

Comment:
Please, check for `child->type == XMPP_STANZA_TAG ` instead of storing string into `child_name`. In this way it will be the same as it is implemented in `xmpp_stanza_get_child_by_name()`.